src/Datadog.Trace/Configuration/GlobalSettings.cs

```diff
@@ -55,6 +77,11 @@ internal GlobalSettings(IConfigurationSource source)
         /// </summary>
         internal bool DiagnosticSourceEnabled { get; }
 
+        /// <summary>
+        /// Gets the factory configurator.
+        /// </summary>
+        internal IFactoryConfigurator FactoryConfigurator { get; private set; }
```

Comment:
I feel that this class should contain "serializable" config. What you add here is a more like "runtime" state of the object created based on settings.